### DISCUSSION TITLE
Quarkus custom extension - build process when config uses Kubernetes ConfigMap & Secrets

### DISCUSSION THREAD

**Discussion by mariuszbozek** [quarkusio/quarkus#31082] [Q&A]
Hi all,

I have a problem with my custom extension. If I don't use Kubernetes ConfigMap and Secrets (everything is configured in the application.properties file) my app with my custom extension works fine. When I configured also Kubernetes ConfigMap and Secrets extension and defined in the application.properties file some entries from K8S Secrets, my app failed because of missing configuration
An example
```
quarkus.my-extension.foo=test <-- ok
quarkus.my-extension.foo=${test-entry-from-k8s} <-- error
```
My custom extension has configured BUILD_AND_RUN_TIME_FIXED config phase for my custom configuration

In the application log a saw that my extension is running before K8s config extension

Is there any chance to check (if k8s config feature is available) and build app in the right order?

[a comment by an automated account is not shown]

**Reply by Sgitario** [reply]
I don't follow what your question is. A full example that we could check might help. 

However, I guess with "Kubernetes ConfigMap and Secrets extension", do you mean the `quarkus-kubernetes-config` extension? 

If what you're asking is how you can check whether the `quarkus-kubernetes-config` extension is enabled, you can use the `quarkus-kubernetes-config` capability is enabled (see documentation [here](https://quarkus.io/guides/writing-extensions#capabilities)). 

Or if you're asking how you can validate your configuration regardless of the extensions you have set up, [this](https://quarkus.io/guides/config-mappings#validation) is how you can add this kind of validation to your application.

About how to build your extension in the right order, you would need to consume a build item that the extension you want to depend on is producing (simply add the build item that the other extension produces as a parameter in your build step). See more [here](https://quarkus.io/guides/writing-extensions#build-items).
